.github/workflows/docker_build.yml

```diff
@@ -84,12 +89,44 @@ jobs:
           username: ${{ secrets.REGISTRY_USERNAME }}
           password: ${{ secrets.REGISTRY_PASSWORD}}
 
-      - name: Build and push
+      - name: Build
+        uses: docker/build-push-action@v2
+        with:
+          context: ${{ inputs.docker_context }}
+          file: ${{ inputs.docker_file }}
+          builder: ${{ steps.buildx.outputs.name }}
+          load: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: |
+            OMNI_COMPONENT=${{ env.COMPONENT_NAME }}
+            NODE_ENV=${{ inputs.node_env }}
+            NPM_TOKEN=${{ secrets.NPM_TOKEN }}
+            OMNI_COMPONENT_COMMIT=${{ github.sha }}
+            OMNI_COMPONENT_VERSION=${{ steps.meta.outputs.version }}
+      
+      - name: Set docker image tar version
+        id: setimgver
+        run: |
+          calculatedSha=$(git rev-parse --short ${{ github.sha }})
+          echo "::set-output name=IMAGE_TAR_VER::$calculatedSha"
+          
+      - name: save docker image to tar file
+        run: docker save --output ${{ inputs.project }}-${{ env.COMPONENT_NAME }}-${{ steps.setimgver.outputs.IMAGE_TAR_VER }}.tar ${{ steps.meta.outputs.tags }}
+
+      - name: upload docker image to private minio storage
+        run: |   
+          export AWS_ACCESS_KEY_ID=${{ secrets.MINIO_CN_ACCESS_KEY_ID }}
+          export AWS_SECRET_ACCESS_KEY=${{ secrets.MINIO_CN_SECRET_ACCESS_KEY }}
+          aws --endpoint-url http://10.10.1.30:9000 s3 cp  "${{ inputs.project }}-${{ env.COMPONENT_NAME }}-${{ steps.setimgver.outputs.IMAGE_TAR_VER }}.tar" s3://docker-img-artifacts 
+
+      - name: push image
         uses: docker/build-push-action@v2
         with:
           context: ${{ inputs.docker_context }}
           file: ${{ inputs.docker_file }}
-          push: ${{ inputs.push }}
+          builder: ${{ steps.buildx.outputs.name }}
+          push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
           build-args: |
```

Comment:
also with `outputs` you don't need to build it again. you can get the tar file and push the image to registry at the same time